textalyze.js

```diff
@@ -1,23 +1,30 @@
-/**
- * This is the base code for v0.1 of our JavaScript text analyzer.
- * Visit https://github.com/jfarmer/text-analysis to see what to do.
- *
- * Send a message in Slack if you're stuck or unsure what to do.  These
- * comments are here to help you, but please delete them as you go along.
- */
-
 /**
  * Given an input Array, returns a Map containing the count of each item in the input.
  * @param {Array} array - The array of items to count
  * @param {Map} counts - A Map containing the counts of the items in the input array
  */
 function itemCounts(array) {
-  let counts = new Map();
+  const counts = new Map();
 
-  // Your code here.
-  // Run 'npm test' to see what tests need to pass.
+  array.forEach((value) => {
+    const count = counts.get(value) || 0;
+    counts.set(value, count + 1);
+  });
 
   return counts;
 }
 
 module.exports = { itemCounts };
+
+
+//
+// running the app
+//
+
+const array = ['a', 'b', 'b', 'c', 'c', 'a', 'd'];
```

Comment:
This code is meant to both be included as a module (via `require`) and run directly (via `npm start`).

To ensure that this code will run _only_ when `textalyze.js` is run directly, do the following: https://gist.github.com/miguelmota/a6258318a5b7ce9b462243e25c2025a5

So, it'd look like:

```javascript
if (require.main === module) {
  const array = ['a', 'b', 'b', 'c', 'c', 'a', 'd'];
  console.log(`...`);
  // etc. etc.
 
}
```

This will prevent the code from being run when `textalyze.js` is loaded via `require`.